app/components/behandeling-van-agendapunt.js

```diff
@@ -1,7 +1,96 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+import { action } from "@ember/object";
+import { inject as service } from '@ember/service';
 
 export default class BehandelingVanAgendapuntComponent extends Component {
-  @tracked title;
   @tracked openbaar;
+  @tracked document
+  @tracked documentContainer;
+  @service store;
+  @service currentSession;
+  @tracked behandeling;
+  editor;
+
+  constructor() {
+    super(...arguments);
+    this.openbaar = this.args.behandeling.openbaar;
+    this.behandeling = this.args.behandeling;
+    this.documentContainer = this.args.behandeling.documentContainer;
+    this.tryToFetchDocument.perform();
+  }
+  @task
+  *tryToFetchDocument() {
+    yield this.args.behandeling.documentContainer;
+    if(this.documentContainer.content) {
+      this.document = yield this.documentContainer.get('currentVersion');
+      if(!this.document) {
+        this.document = this.store.createRecord('editor-document');
+      }
+    } else {
+      this.document = this.store.createRecord('editor-document');
+      this.documentContainer = this.store.createRecord('document-container');
+      this.documentContainer.currentVersion = this.document;
+      this.behandeling.documentContainer = this.documentContainer;
+      yield this.document.save();
+      yield this.documentContainer.save();
+      yield this.behandeling.save();
+    }
+  }
+  @action
+  async save(e) {
+    e.stopPropagation();
+    this.behandeling.openbaar = this.openbaar;
+    await this.saveEditorDocument.perform(this.document);
```

Comment:
we should at least have some error handling here and not update the behandeling if the save of the document failed